### DISCUSSION TITLE
Problem to access EJBCA Administration

### DISCUSSION THREAD

**Discussion by aleng1** [Keyfactor/ejbca-ce#450] [Q&A]
Hi everyone, im facing problem to access URL with this docker-compose.yml:
version: "3"
networks:
    access-bridge:
        driver: bridge
    application-bridge:
        driver: bridge
services:
    ejbca-database:
        container_name: ejbca-database
        image: "library/mariadb:latest"
        networks:
            - application-bridge
        environment:
            - MYSQL_ROOT_PASSWORD=foo123
            - MYSQL_DATABASE=ejbca
            - MYSQL_USER=ejbca
            - MYSQL_PASSWORD=ejbca
        volumes:
            - ./datadbdir:/var/lib/mysql:rw
    ejbca-node1:
        hostname: ejbca-node1
        container_name: ejbca
        image: keyfactor/ejbca-ce:latest
        depends_on:
            - ejbca-database
        networks:
            - access-bridge
            - application-bridge
        environment:
            - DATABASE_JDBC_URL=jdbc:mariadb://ejbca-database:3306/ejbca?characterEncoding=UTF-8
            - LOG_LEVEL_APP=INFO
            - LOG_LEVEL_SERVER=INFO
            - TLS_SETUP_ENABLED=simple
        ports:
            - "80:8080"
            - "443:8443"

When i open up the URL in browser the connection cannot be made. Btw im using Windows environment

Sorry to ask silly question, im still new to this technology

**Comment by primetomas** [comment]
Does the log say anything, for example problems connecting to the database?

**Reply by aleng1** [reply]
yes the logs shows this:
`ejbca           | 2023-12-20 01:21:20,213+0000 INFO  [org.apache.commons.beanutils.FluentPropertyBeanIntrospector] (main) Error when creating PropertyDescriptor for public final void org.apache.commons.configuration2.AbstractConfiguration.setProperty(java.lang.String,java.lang.Object)! Ignoring this property.
ejbca           | 2023-12-20 01:21:20,514+0000 ERROR [org.ejbca.ui.cli.jdbc.JdbcTool] (main) Socket fail to connect to host:address=(host=ejbca-database)(port=3306)(type=primary). Connection refused (Connection refused)
ejbca           | 2023-12-20 01:21:20,514+0000 INFO  [org.ejbca.ui.cli.jdbc.JdbcTool] (main) Done.
ejbca           | 2023-12-20 01:21:20,547+0000 INFO  [/opt/keyfactor/bin/start.sh] (process:1) Waiting for external database 'jdbc:mariadb://ejbca-database:3306/ejbca?characterEncoding=UTF-8' to become available.
ejbca           | 2023-12-20 01:21:24,924+0000 INFO  [org.apache.commons.beanutils.FluentPropertyBeanIntrospector] (main) Error when creating PropertyDescriptor for public final void org.apache.commons.configuration2.AbstractConfiguration.setProperty(java.lang.String,java.lang.Object)! Ignoring this property.
ejbca           | 2023-12-20 01:21:25,114+0000 ERROR [org.ejbca.ui.cli.jdbc.JdbcTool] (main) Socket fail to connect to host:address=(host=ejbca-database)(port=3306)(type=primary). Connection refused (Connection refused)
ejbca           | 2023-12-20 01:21:25,115+0000 INFO  [org.ejbca.ui.cli.jdbc.JdbcTool] (main) Done.`

any suggestion how to fix it? tried to restart mysql service but still fail to connect to host

**Comment by primetomas** [comment]
The error is simple:
"Socket fail to connect to host:address=(host=ejbca-database)(port=3306)(type=primary). Connection refused"
The EJBCA container can not connect to the database. There is some firewall blocking, or the database is not listening to external ports? Or

**Reply by aleng1** [reply]
Thank you for replying. The problem has been solved as I'm using Linux VM to follow the tutorial. One more thing, the link to access the admin web is https://localhost/ejbca/adminweb/, not https://ejbca-node1/ejbca/adminweb/. Everything is okay now.
